builder/frameworks/fsp.py

```diff
@@ -0,0 +1,103 @@
+from os.path import isdir, join
+
+from SCons.Script import DefaultEnvironment
+
+env = DefaultEnvironment()
+platform = env.PioPlatform()
+board = env.BoardConfig()
+
+env.SConscript("_bare.py")
+
+FRAMEWORK_DIR = platform.get_package_dir("framework-renesas-fsp")
+assert isdir(FRAMEWORK_DIR)
+
+env.Append(
+    CFLAGS=[
+        "-std=gnu11"
+    ],
+
+    CPPDEFINES=[
+        "_RENESAS_RA_"
+    ],
+
+    CXXFLAGS=[
+        "-std=gnu++17",
+        "-fno-use-cxa-atexit"
+    ],
+
+    CPPPATH=[
+        join(FRAMEWORK_DIR, "fsp", "inc"),
+        join(FRAMEWORK_DIR, "fsp", "inc", "api"),
+        join(FRAMEWORK_DIR, "fsp", "inc", "instances"),
+        join(FRAMEWORK_DIR, "fsp", "src"),
+        join(FRAMEWORK_DIR, "fsp", "src", "r_sce"),
+        join(FRAMEWORK_DIR, "fsp", "src", "r_sce", "common"),
+        join(FRAMEWORK_DIR, "fsp", "inc", "arm", "CMSIS_5", "CMSIS", "Core", "Include"),
+        join(FRAMEWORK_DIR, "variants", board.get("build.variant")),
+        join(FRAMEWORK_DIR, "variants", board.get("build.variant"), "includes", "ra_gen"),
+        join(FRAMEWORK_DIR, "variants", board.get("build.variant"), "includes", "ra_cfg", "fsp_cfg"),
+        join(FRAMEWORK_DIR, "variants", board.get("build.variant"), "includes", "ra_cfg", "fsp_cfg", "bsp"),
+        join(FRAMEWORK_DIR, "variants", board.get("build.variant"), "tmp_gen_c_files")
+    ],
+    LINKFLAGS=[
+        "--specs=nano.specs"
+    ],
+    LIBPATH=[
+        join(FRAMEWORK_DIR, "variants", board.get("build.variant"))
+    ],
+    LDSCRIPT_PATH=join(FRAMEWORK_DIR, "variants", board.get("build.variant"), "fsp.ld")
+)
+
+libs = []
+
+# Extracted from ArduinoCore-Renesas\extras\e2studioProjects\Santiago\configuration.xml "raComponentSelection"
+to_be_built_sublibs_uno_r4_x = ["r_adc", "r_agt", "r_cgc", "r_dac", "r_doc", "r_dmac", "r_dtc", "r_elc", "r_gpt", "r_icu", "r_iic_master", "r_iic_slave", "r_ioport", "r_kint",
+                       "r_lpm", "r_rtc", "r_sce/crypto_procedures/src/sce5", "r_sce5_ra4", "r_sci_i2c", "r_sci_spi", "r_sci_uart", "r_spi", "r_usb_basic", "r_usb_pcdc", "r_wdt", "r_can", "r_flash_lp", "rm_vee_flash"]
+# Extracted from ArduinoCore-Renesas\extras\e2studioProjects\portenta_h33_lib
+to_be_built_sublibs_portenta_c33 = ["r_adc", "r_agt", "r_canfd", "r_cgc", "r_crc", "r_dac", "r_dmac", "r_dtc", "r_elc", "r_ether_phy", "r_ether", "r_flash_hp", "r_gpt", "r_icu", "r_iic_master", "r_iic_slave",
+                                    "r_ioport", "r_lpm", "r_qspi", "r_rtc", "r_sce_protected/crypto_procedures/src/sce9", "r_sci_i2c", "r_sci_spi", "r_sci_uart", "r_spi", "r_ssi", "r_usb_basic", "r_usb_pcdc", "r_usb_composite", "r_wdt", "rm_adpcm_decoder", "r_sdhi"]
+# This is always built
+src_filter = "-<*> +<bsp/cmsis> +<bsp/mcu/all> "
+if str(board.get("build.mcu", "")).lower().startswith("ra4m1"):
```

Comment:
The `str()` call is probably not necessary as the `mcu` field is already a string.

Reply:
`str()` removed, still works.